integration/run_test.go

```diff
@@ -468,3 +468,178 @@ func TestKritisPods(t *testing.T) {
+		shouldSucceed        bool
+		shouldLabel          bool
+
+		remoteOnly bool
+		cleanup    func(t *testing.T)
+	}
+
+	var testCases = []testRunCase{
+		{
+			description: "nginx-no-digest-breakglass",
+			args: []string{"kubectl", "apply", "-f",
+				"integration/testdata/nginx/nginx-no-digest-breakglass.yaml"},
+			pods: []testObject{
+				{
+					name: "nginx-no-digest-breakglass",
+				},
+			},
+			shouldSucceed: true,
+			shouldLabel:   true,
+			dir:           "../",
+			cleanup: func(t *testing.T) {
+				cmd := exec.Command("kubectl", "delete", "-f",
+					"integration/testdata/nginx/nginx-no-digest-breakglass.yaml")
+				cmd.Dir = "../"
+				output, err := integration_util.RunCmdOut(cmd)
+				if err != nil {
+					t.Fatalf("kubectl delete failed: %s %v", output, err)
+				}
+			},
+		},
+		{
+			description: "image-with-acceptable-vulnz",
+			args: []string{"kubectl", "apply", "-f",
+				"integration/testdata/vulnz/acceptable-vulnz.yaml"},
+			pods: []testObject{
+				{
+					name: "image-with-acceptable-vulnz",
+				},
+			},
+			shouldSucceed: true,
+			shouldLabel:   false,
+			dir:           "../",
+			cleanup: func(t *testing.T) {
+				cmd := exec.Command("kubectl", "delete", "-f",
+					"integration/testdata/vulnz/acceptable-vulnz.yaml")
+				cmd.Dir = "../"
+				output, err := integration_util.RunCmdOut(cmd)
+				if err != nil {
+					t.Fatalf("kubectl delete failed: %s %v", output, err)
+				}
+			},
+		},
+	}
+
+	ns, deleteNs := setupNamespace(t)
+	defer deleteNs()
+	createGACSecret(t, ns)
+	deleteKritis := initKritis(t, ns)
+	defer deleteKritis()
+	if err := kubernetesutil.WaitForDeploymentToStabilize(client, ns.Name,
+		fmt.Sprintf("kritis-validation-hook-%s", ns.Name), 2*time.Minute); err != nil {
+		t.Fatalf("Timed out waiting for deployment to stabilize")
+	}
+	createCRDExamples(t, ns)
+	waitForCRDExamples(t, ns)
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			defer testCase.cleanup(t)
+
+			cmd := exec.Command(testCase.args[0], testCase.args[1:]...)
+			cmd.Dir = testCase.dir
+			output, err := integration_util.RunCmdOut(cmd)
+			if err != nil {
+				if !testCase.shouldSucceed {
+					return
+				}
+				t.Fatalf("testCase cmd failed: %s %v\n%s", output,
+					err, getKritisLogs(t))
+
+			}
+			if !testCase.shouldSucceed {
+				t.Fatalf("deployment should have failed but succeeded\n%s",
+					getKritisLogs(t))
+			}
+
+			for _, p := range testCase.pods {
+				if err := kubernetesutil.WaitForPodReady(client.CoreV1().Pods(ns.Name), p.name); err != nil {
+					t.Fatalf("Timed out waiting for pod ready\n%s\n%s",
+						getKritisLogs(t),
+						output)
+				}
+			}
+
+			for _, d := range testCase.deployments {
+				if err := kubernetesutil.WaitForDeploymentToStabilize(client, ns.Name, d.name, 10*time.Minute); err != nil {
+					t.Fatalf("Timed out waiting for deployment to stabilize\n%s",
+						getKritisLogs(t))
+				}
+				if testCase.deploymentValidation != nil {
```

Comment:
sorry if i'm missing something, but in what scenario would you use deploymentValidation?

Reply:
this is an additional feature for future integration tests in which you can add a function to verify a deployment as sometimes all pods "READY" is not enough